packages/forklift-console-plugin/src/modules/Mappings/mappingActions.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useMemo } from 'react';
+import { ConfirmModal } from 'legacy/src/common/components/ConfirmModal';
+import { AddEditMappingModal } from 'legacy/src/Mappings/components/AddEditMappingModal';
+import { useDeleteMappingMutation } from 'legacy/src/queries';
+import { Mapping, MappingType } from 'legacy/src/queries/types';
+import { useTranslation } from 'src/utils/i18n';
+
+import { withActionContext } from '@kubev2v/common/components/ActionServiceDropdown';
+import withQueryClient from '@kubev2v/common/components/QueryClientHoc';
+import { useModal } from '@kubev2v/common/polyfills/sdk-shim';
+
+import { CommonMapping } from './dataCommon';
+import { FlatNetworkMapping } from './dataForNetwork';
+import { FlatStorageMapping } from './dataForStorage';
+
+export const useNetworkMappingActions = ({ entity }: { entity: FlatNetworkMapping }) =>
+  useMappingActions<FlatNetworkMapping>({ entity, mappingType: MappingType.Network });
+
+export const useStorageMappingActions = ({ entity }: { entity: FlatStorageMapping }) =>
+  useMappingActions<FlatStorageMapping>({ entity, mappingType: MappingType.Storage });
+
+function useMappingActions<T extends CommonMapping>({
+  entity,
+  mappingType,
+}: {
+  entity: T;
+  mappingType: MappingType;
+}) {
+  const { t } = useTranslation();
+  const launchModal = useModal();
+  const areProvidersReady = entity.sourceReady && entity.targetReady;
+  const areProvidersResolved = entity.sourceResolved && entity.targetResolved;
+  const editingDisabled = !areProvidersReady || !areProvidersResolved;
```

Comment:
is it documented ?
what is the reason for not allowing to edit a mapping that contain none ready providers ?

doesn't it makes sense to allow edit it to update the provider ?

Reply:
@yaacov 
> is it documented ?
>
Not sure. I will check.

> what is the reason for not allowing to edit a mapping that contain none ready providers ?
> doesn't it makes sense to allow edit it to update the provider ?
>
The mapping uses inventory data (networks/storages). If inventory is re-building then we might have inconsistent state i.e. partially old and partially new networks. The backend should handle that but disabling editing makes sense from usability point of view.